packages/js/core/src/types/UriRedirect.ts

```diff
@@ -1,14 +1,14 @@
-import { Uri, PluginPackage } from ".";
+import { PluginPackage } from ".";
 
 /**
  * Redirect from one URI, or a set of URIs, to a new URI or a plugin.
  */
 export interface UriRedirect {
   /** Redirect from this URI */
-  from: Uri;
+  from: string;
```

Comment:
I think that keeping the underlying `UriRedirect` type the same is useful. Here's why:
- Switching to strings is purely to help keep the user/developer facing interface simple and easy to use.
- Having the backing type remain a `Uri` is useful so we can be sure that we're operating on a "correct" URI string that's been sanitized.

In the current implementation you have, a user wouldn't know if their provided string is an incorrect URI until it gets converted into a `new Uri(userString)` somewhere in the algorithms at run-time. This is bad sanitization.

We should instead be converting the strings to `new Uri`s within the Web3ApiClient constructor, so that we can catch bad URI strings right off the bat.

Reply:
Created new type `UriRedirectDefinition` that has from and to as string which is what users will need to pass into client and then converted to UriRedirect in constructor.
Is this more in line with what you had in mind?

Reply:
Yes that sounds great